src/bindings/python/tests/test_runtime/test_core.py

```diff
@@ -98,6 +98,28 @@ def test_compile_model_without_device(request, tmp_path):
     assert isinstance(compiled_model, CompiledModel)
 
 
+# request - https://docs.pytest.org/en/7.1.x/reference/reference.html#request
+def test_compile_model_with_auto_plugin(request, tmp_path):
+    core = Core()
+    xml_path, bin_path = create_filename_for_test(request.node.name, tmp_path)
+    relu_model = get_relu_model()
+    serialize(relu_model, xml_path, bin_path)
+    model = core.read_model(model=xml_path, weights=bin_path)
```

Comment:
these lines may be deleted-- we can past to `compile_model` just `relu_model`

Reply:
moreover I'm not sure do we need such a test -- actually we already [have similar](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/tests/test_runtime/test_core.py#L34).
And [another one](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/tests/test_runtime/test_core.py#L44) with path argument

Reply:
I would even combine tests with parametrization and rename/delete some of them if necessary. For example: `test_compile_model_with_auto_plugin` and `test_compile_model_with_auto_plugin_no_default` are indeed really similar to other ones.

Let it be something in lines of:
```python
@pytest.mark.parametrize(("loading_type"), [
    None,
    "posixpath",
    ...
])
@pytest.mark.parametrize("device_name", [
    None,
    "CPU",
    ...
])
@pytest.mark.parametrize("config", [
    None,
    {"PERFORMANCE_HINT": "THROUGHPUT"},  # here I would replace them with actual properties later CC Ana
    ...
])
def test_compact_api(request, tmp_path, loading_type, device_name, config):
    core = Core()
    
    model = get_relu_model()
    
    if loading_type is not None:
        xml_path, bin_path = create_filename_for_test(request.node.name, tmp_path)
        serialize(relu_model, xml_path, bin_path)
        model = core.read_model(model=xml_path, weights=bin_path)

    compiled_model = None  # placeholder value
    
    if device_name is None:
        compiled_model = compile_model(model=model, config=config)
    
    compiled_model = compile_model(model=model, device_name=device_name, config=config)
    
    assert isinstance(compiled_model, CompiledModel)
```
It needs some refinement and catching corner cases. However, overall the idea is to squeeze similar tests together, as they are testing the same thing but only with different parameters.

@siddhant-0707 @akuporos @p-wysocki let me know what you think about this.